pkg/reconciler/ingress/resources/gateway.go

```diff
@@ -395,24 +396,47 @@ func GetNonWildcardIngressTLS(ingressTLS []v1alpha1.IngressTLS, nonWildcardSecre
 	return result
 }
 
-// GetIngressGatewaySvcNameNamespaces gets the Istio ingress namespaces from ConfigMap.
-// TODO(nghia):  Remove this by parsing at config parsing time.
-func GetIngressGatewaySvcNameNamespaces(ctx context.Context) ([]metav1.ObjectMeta, error) {
-	cfg := config.FromContext(ctx).Istio
-	nameNamespaces := make([]metav1.ObjectMeta, len(cfg.IngressGateways))
-	for i, ingressgateway := range cfg.IngressGateways {
-		parts := strings.SplitN(ingressgateway.ServiceURL, ".", 3)
-		if len(parts) != 3 {
-			return nil, fmt.Errorf("unexpected service URL form: %s", ingressgateway.ServiceURL)
-		}
-		nameNamespaces[i] = metav1.ObjectMeta{
-			Name:      parts[0],
-			Namespace: parts[1],
+// GetIngressGatewaySvcNameNamespaces gets the Istio ingress namespaces from ConfigMap for gateways that should expose the service.
+func GetIngressGatewaySvcNameNamespaces(ctx context.Context, obj kmeta.Accessor) ([]metav1.ObjectMeta, error) {
```

Comment:
It seems to me that wherever we have a kmeta.Accessor we could replace it with `ing *v1alpha1.Ingress`, correct?

Reply:
True. I used to have `ing *v1alpha1.Ingress` & @dprotaso made me change for this. Which is imho better to depends on a small interface rather than an implementation

See https://github.com/knative-extensions/net-istio/pull/1250#discussion_r1496262747

Reply:
It depends generalizing for one case it is still weird but maybe a good fir for the GetLabels method.